src/lib.rs

```diff
@@ -915,8 +949,10 @@ pub fn render_opts<'a,
 
     for e in g.edges().iter() {
         let escaped_label = &g.edge_label(e).to_dot_string();
-        let start_arrow = &g.edge_start_arrow(e).to_dot_string();
-        let end_arrow = &g.edge_end_arrow(e).to_dot_string();
+        let start_arrow = g.edge_start_arrow(e);
+        let end_arrow = g.edge_end_arrow(e);
+        let start_arrow_s = start_arrow.to_dot_string();
+        let end_arrow_s = end_arrow.to_dot_string();
```

Comment:
These four lines are a bit awkward, but I will address this after this PR by reorienting all the code to be Display-oriented and using composable formatting instead of allocating temporary Strings.